Sources/Swift/Tools/UIRedactBuilder.swift

```diff
@@ -141,46 +133,65 @@ class UIRedactBuilder {
         return image.imageAsset?.value(forKey: "_containingBundle") == nil
     }
     
-    private func mapRedactRegion(fromView view: UIView, to: CALayer, redacting: inout [RedactRegion], area: CGRect, redactText: Bool, redactImage: Bool) {
-        let rectInWindow = (view.layer.presentation() ?? view.layer).convert(view.bounds, to: to)
-        guard (redactImage || redactText) && area.intersects(rectInWindow) && !view.isHidden && view.alpha != 0 else { return }
+    private func mapRedactRegion(fromView view: UIView, redacting: inout [RedactRegion], rootFrame: CGRect, redactOptions: SentryRedactOptions, transform: CGAffineTransform) {
+        guard (redactOptions.redactAllImages || redactOptions.redactAllText) && !view.isHidden && view.alpha != 0 else { return }
+        
+        let layer = view.layer.presentation() ?? view.layer
+        let size = layer.bounds.size
+        let layerMiddle = CGPoint(x: size.width * layer.anchorPoint.x, y: size.height * layer.anchorPoint.y)
+        
+        var newTransform = transform.translatedBy(x: layer.position.x, y: layer.position.y)
+        newTransform = view.transform.concatenating(newTransform)
+        newTransform = newTransform.translatedBy(x: -layerMiddle.x, y: -layerMiddle.y)
         
         let ignore = shouldIgnore(view: view)
-        let redact = shouldRedact(view: view, redactText: redactText, redactImage: redactImage)
+        let redact = shouldRedact(view: view, redactOptions: redactOptions)
         
         if !ignore && redact {
-            redacting.append(RedactRegion(rect: rectInWindow, color: self.color(for: view)))
+            redacting.append(RedactRegion(size: size, transform: newTransform, type: .redact, color: self.color(for: view)))
             return
-        } else if hasBackground(view) {
-            if rectInWindow == area {
+        } else if isOpaque(view) {
+            let finalViewFrame = CGRect(origin: .zero, size: size).applying(newTransform)
+            if isAxisAligned(newTransform) && finalViewFrame == rootFrame {
+                //Because the current view is covering everything we found so far we can clear `redacting` list
                 redacting.removeAll()
             } else {
-                redacting = redacting.flatMap { $0.splitBySubtracting(region: rectInWindow) }
+                redacting.append(RedactRegion(size: size, transform: newTransform, type: .clip))
```

Comment:
I tried to create an example that uses clip regions, but I'm not quite sure what it means still or what the outcome is. I figured it is something like, there's a redacted region, but then something inside it actually doesn't need to be redacted, so place a mask over it to essentially punch a hole out of the redaction region. Is that right? Or is this the actual crux of this PR, where the redaction region is the bounding box containing a transformed view (like the rotated text), and we don't want to redact the whole bounding box, just the view itself, by created a rectangle of the same size and transformation?

Reply:
You're right, the clip (with became [clipOut](https://github.com/getsentry/sentry-cocoa/pull/4325)) is responsible to create a hole in every redact region that comes after it